Add and test Reducers

RayRedGoose opened the pull request:
### This PR contains the next changes:

- Updated interfaces: fixing errors, adding readonly type for not changeable data, adding actions property to AppStore interface.
- Updated connection with Redux: adding correct check type with AppStore interface.
- Added all reducers to root reducer based on AppStore interface.
- Added structure for every reducer
- Added test for reducers

### Tests

- `git fetch`
- `git checkout reducers`
- `npm test`
- 29 tests for reducers should be passed.

Resolves #26

JamesRexMiller4 commented:
Nice work on writing out all the interfaces, and the tests for the reducers. This will make dispatching to the store a breeze and we can focus on the component functionality and logic. Great job!

RayRedGoose commented:
I added action creators for updating app store, also I tested all action creators. After adding new actions I updated reducer and added new test for new cases.
